perses/dispersed/loopy_annealing_integrator.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+
+from openmmtools.integrators import AlchemicalNonequilibriumLangevinIntegrator
+from simtk import unit
+import numpy as np
+import simtk.openmm as openmm
+from openmmtools.constants import kB
+from openmmtools import utils
+
+
+# In[119]:
+
+
+class LoopyAlchemicalNonequilibriumLangevinIntegrator(AlchemicalNonequilibriumLangevinIntegrator):
+    """
+    Subclass of `AlchemicalNonequilibriumLangevinIntegrator` that hardcodes the following integration scheme:
+
+    Step 1: lambda_0 equilibrium sampling
+        run nsteps_eq of BAOAB (V R O R V) integration at the alchemical lambda_0 endstate
+    Step 2: Forward Annealing
+        run n_steps_neq of BAOAB (V R O R V H); 'H': Hamiltonian update step
+    Step 3: lambda_1 equilibrium sampling
+        run Step 1 at the lambda_1 endstate
+    Step 3: Reverse Annealing
+        run Step 2 in the reverse direction
+
+    forward and backward works are saved to `forward_work` and `backward_work`, respectively
+    """
+    def __init__(self,
+                 alchemical_functions = None,
+                 nsteps_eq = 1000,
+                 nsteps_neq = 100,
+                 **kwargs):
+        """        
+        arguments
+            nsteps_eq : int
+                number of equilibration steps to run at either endstate
+            nsteps_neq : int
+                number of nonequilibrium annealing steps to run in either direction
+
+        parameters
+            _n_steps_eq : int
+                number of BAOAB loops to run at each endstate in an attempt to generate i.i.d samples from endstates
+        """
+        self._n_steps_eq = nsteps_eq
+        splitting = 'V R O R V'
+        super().__init__(alchemical_functions, splitting, nsteps_neq = nsteps_neq, **kwargs)
+    
+    def _add_global_variables(self):
+        """
+        modify the super (_add_global_variables) to add a collector for forward and reverse annealing, as well as to specify the number of equilibration
+        steps at endstates.
+        """
+        super()._add_global_variables()
+        self.addGlobalVariable('forward_work', 0)
+        self.addGlobalVariable('backward_work', 0)
+        self.addGlobalVariable('n_steps_eq', self._n_steps_eq)
```

Comment:
Should we be consistent between the class arguments (`nsteps_eq`, `nsteps_neq`) and the global parameter names (`n_steps_eq`, `n_steps_neq`)?

Reply:
I guess this may not matter much since we call the nonequilibrium switching time something complicated like `n_lambda_steps`

Reply:
this is how `AlchemicalNonequilibrumLangevinIntegrator` treats `nsteps_neq`; i figured i'd preserve the internal consistency defined in the super class